### DISCUSSION TITLE
Slowness boot of iPXE

### DISCUSSION THREAD

**Discussion by Yuvaraj2021** [ipxe/ipxe#493] [Q&A]
Dear Friends,

During iPXE boot, "Downloading NBP file...." taking 10 to 15 mins. to download and boot iPXE ...

Please share your valuable experience to fix this slowness issues.

![iPXE_Boot](https://user-images.githubusercontent.com/92423015/137109878-94fa8d63-93c9-4ea1-8b49-f77b07c328d3.jpg)

![iPXE](https://user-images.githubusercontent.com/92423015/137108879-af7ce263-9ecb-4cd1-bd24-0ad71c90399e.JPG)

**Comment by NiKiZe** [comment]
Download NBP is your EFI firmware - before iPXE
Check your link speed, maybe it is 10 mbit or other hardware issues with the link. (link settings missmatch as an example.)
Also check if there is any firmware update available for your system.

What you can do to mitigate the issue is try to build a reduced version of iPXE with just the needed driver, Smaller size = shorter time.

Is both pictures from the same machine?

**Reply by Yuvaraj2021** [reply]
Thanks a lot for your prompt response. 

Yes, both snaps are captured from same machine. Sure, I will check the link speed.

If we use lower version of iPXE, slowness issues will be fixed ?

**Reply by NiKiZe** [reply]
If your 10-15 minute wait is between "Downloading NBP file ..." and "NBP file downloaded successfully." then your issue is related to your firmware, NIC, or network - not iPXE since it is before iPXE has even started.

But by using a ipxe build that is smaller in filesize (using snponly.efi instead of ipxe.efi) you can reduce the time taken, this is not a fix, but rather a workaround until you can figure out what is actually going on.

I would check network, and TFTP server configuration, and then use tcpdump to capture the tftp data, maybe you have some bad configuration, lots of retries or similar.
